### PR TITLE
refactor: prevent direct publisher calls

### PR DESCRIPTION
**Pull request by loks0n**
- **update: use new terminologies on path params and local variables.**
- **add: filter for latest changes.**
- **fix|update: database tests for new error messages.**
- **add: todo.**
- **change: events system.**
- **fix: tests as per new events system.**
- **fix: graphql tests.**
- **fix: graphql tests.**
- **fix: graphql tests, again.**
- **init: move to module structure for database; start with columns.**
- **update: move the init hook.**
- **update: move DB ops to module.**
- **update: move Table ops to module.**
- **update: move Index ops to module.**
- **fix: imports, lint.**
- **add: register index and table modules.**
- **add: move and register row modules.**
- **add: usage.**
- **remove: database controller 👋**
- **add: database module to platform registry.**
- **update: keep message same.**
- **update: error messages.**
- **fix: `float` column creation!**
- **fix: lint!**
- **update: `collection` response model to `table`.**
- **update: `attribute*` response model to `column*`.**
- **update: `document` response model to `row`; refactor: old collection to table models on response types.**
- **update: return keys.**
- **update: services.**
- **fix: response modals, mappings, filters.**
- **patch: index module to use/specify `columns` instead of `attributes`.**
- **remove: old attributes anyway.**
- **update: tests to use the 1.7.x changes on databases.**
- **wip: events compat based on version.**
- **fix: project usage `documents` to `rows`.**
- **fix: project and general usage tests.**
- **fix: migrations test <> csv imports.**
- **fix: graphql tests.**
- **update: tests.**
- **remove: filter.**
- **fix: realtime channels; update: row model.**
- **update: request, response filters.**
- **fix: filtering**
- **remove: aliasing filter.**
- **add: backported channels.**
- **update: `collections` <> `tables` errors.**
- **update: `documents` <> `rows` errors.**
- **update: `attributes` <> `columns` errors.**
- **update: rename remaining variables.**
- **update: change filter name for collections <> tables.**
- **update: change filter name for attributes <> columns.**
- **manage: events and channels.**
- **update: re-add legacy models; remove: filters.**
- **update: keep both tables and collections api errors.**
- **update: events schema, response models for database usages.**
- **update: events, errors, realtime.**
- **misc.**
- **update: abstraction over table and collection apis.**
- **update: handle the context logic, use defaults.**
- **update: abstraction over attribute and column apis [WIP].**
- **update: address minor comments by `coderabbitai`.**
- **update: abstraction over attribute and column apis.**
- **update: abstraction over document and row apis.**
- **update: use `abstract` for `getResponseModel`.**
- **update: use nested directory structure.**
- **add: indexes api to tables.**
- **update: sdk namespaces.**
- **fix: missing path :\**
- **fix: namespaces.**
- **fix: namespaces, again.**
- **fix: another pesky sdk namespace 🤨.**
- **update: split service registrations into smaller registries.**
- **misc: name fixes.**
- **misc: fixes.**
- **update: the usage modals.**
- **update: the usage model.**
- **update: move around tests.**
- **update: fix exception.**
- **fix: wrong param.**
- **misc: changes to the databases worker.**
- **fix: var ref.**
- **misc: fix error types.**
- **add: table tests to usage.**
- **fix: realtime events.**
- **update: short the callbacks.**
- **revert: tests to `collections` API.**
- **fix|add: tables api tests to realtime.**
- **fix: realtime channel count.**
- **add: table api tests.**
- **add: table api tests to abuse.**
- **revert: graphql tests for collections api.**
- **patch: graphql tests <> collections.**
- **update: model; add|update: graphql tests [wip].**
- **update: realtime events logic.**
- **fix: endpoint in abuse test.**
- **update: database server test for graphql <> tables api.**
- **fix: graphql tests.**
- **update: test coverage for databases.**
- **update: simplify setting events context param.**
- **update: fix test variable.**
- **add: table apis to webhook tests.**
- **add: table apis to webhook custom server tests.**
- **address naming comments.**
- **update: exception name.**
- **update: var name.**
- **fix: audits event.**
- **fix: method name.**
- **fix: abuse tests!**
- **remove: todo.**
- **address comments.**
- **update: `getInternalId()` -> `getSequence()`**
- **update: remove conflict, restrict and relationship exceptions from global handler.**
- **update: remove query exceptions from global handler.**
- **update: remove structure exceptions from global handler.**
- **update: remove duplicate exceptions from global handler.**
- **update: remove authorization from global handler.**
- **update: remove not-found and index dependency exceptions from global handler.**
- **remove: `$isTablesAPI` var.**
- **rollback: auth exception as per `1.8.x` source.**
- **add/update: `createDocuments` on module structure.**
- **fix: missing context!**
- **add: upsert for collections and tables.**
- **update: better syntax for callbacks.**
- **fix: duplicate call to collection creation.**
- **fix: string encryption attribute in model.**
- **fix: index lengths.**
- **remove: duplicate events.**
- **fix: index lengths.**
- **add: lengths to column indexes.**
- **add: upsert documents/rows.**
- **update: descriptions.**
- **fix: audit event.**
- **add: delete documents/rows**
- **update: model.**
- **add: update documents/rows.**
- **add: increment, decrement document/row attribute/column; fixes: misc.**
- **fix: route name 😒**
- **fix: a bunch of tests 🚧**
- **fix: one more test before I sign out 👋**
- **fix: one more test...**
- **fix: missing context.**
- **fix: pending tests.**
- **update: naming convention for table tests.**
- **add: graphql tests for bulk apis.**
- **add: graphql tests for bulk apis to server tests.**
- **add: graphql tests for bulk apis to server tests.**
- **update: realtime channels checks for tables api.**
- **remove: filter queries.**
- **added invalidating sessions for the project users**
- **updated description of on password change**
- **added onPasswordChange property to the console project**
- **updated onPasswordChange to invalidate session**
- **updated indentation**
- **updated session invalidation tests , models and descriptions**
- **fix: events logic 😮‍💨**
- **fix: remove event for bulk api support.**
- **revert: stuff that is not supposed to be in this branch.**
- **test: backwards compat for request/response filters.**
- **test: backwards compat for request/response filters on latest `1.6.x`.**
- **revert: sanity check tests!**
- **update: add cached logic to `processDocument`.**
- **update: comment.**
- **updated example of the authInvalidateSession**
- **address comments.**
- **ci: empty commit**
- **update: disable `1.8.x` request/response filters for `lazy-loading-relationships`.**
- **updated invalidation session during the change password in the account endpoint**
- **updated tests and project invaldate session**
- **linting**
- **update: docs.**
- **rename: tests from `collections` to `legacy`.**
- **change: namespace.**
- **update: init with module structure.**
- **remove: `use HTTP`. update: inline context logic.**
- **remove: leftover.**
- **address comments.**
- **address comments.**
- **address comment.**
- **address comment.**
- **revert: legacy tests namespacing to old.**
- **fix: sdk namespacing and graphql tests.**
- **update: generate latest specs.**
- **update: generate docs examples as per latest `1.8.x` source.**
- **fix: graphql test for complex queries.**
- **update: re-add filters since the PR is in draft and will be merged in the end.**
- **update: formatter, specs, docs, sdk method names.**
- **fix: missing range on tables api.**
- **ci: empty commit**
- **fix: oha as per latest updates.**
- **fix: oha as per latest updates.**
- **fix: oha as per latest updates.**
- **fix: oha as per latest updates.**
- **fix: oha as per latest updates.**
- **fix: oha as per latest updates.**
- **chore: deprecate old database methods**
- **chore: add namespace to replacements**
- **chore: add more deprecations, add warning headers**
- **chore: add more deprecations**
- **chore: fix table replacewith methods**
- **add: validators to swagger2.**
- **add: new scopes to the api.**
- **update: doc links.**
- **fix: name.**
- **bump: specs for sdks.**
- **bump: sdk examples.**
- **add: scopes on migrations.**
- **add: scopes on migrations and update tests.**
- **fix: namespace for table logs.**
- **update: docs, misc.**
- **update: docs, misc.**
- **update: examples.**
- **remove: wrong examples.**
- **bump: dependencies.**
- **updated the db version**
- **empty commit**
- **chore: add deprecated version**
- **chore: update ordering**
- **chore: remove deprecatedMessage**
- **chore: update format**
- **chore: move deprecated**
- **fix: replacewith methods**
- **added response filters for version older than 1.8.x**
- **add: changes to module structure.**
- **update: mirror tests to Tables api.**
- **bump: dependencies.**
- **update: response modal.**
- **update: log message with fallbacks.**
- **update: tests.**
- **fix: `null` permissions.**
- **fix: `null` permissions on tables api as well.**
- **bump: specs.**
- **update: maintain consistency for permissions with a null check.**
- **update: maintain tests consistency.**
- **update: code quality.**
- **Bump version**
- **fix: the default templates.**
- **fix: session alerts template.**
- **fix: add a default client name for session alerts**
- **fix: address comment.**
- **Update console**
- **Update libs**
- **Revert DB**
- **Update tables test**
- **Downgrade migrations**
- **Check tests**
- **Check tests**
- **bump: response format for preview sdk.**
- **added the cases of null permissions in the upsert route and update the e2e test**
- **Fallback index length 0**
- **Remove var_dump**
- **lock**
- **formatting**
- **Remove var_dump**
- **fix: Use direct source for file-preview when empty**
- **Update libs**
- **Fix index tests**
- **FIx name mismatch**
- **Fix index tests**
- **Update database**
- **fix: add wildcards on `getDocument` response model as well.**
- **Fix index tests**
- **Update database**
- **Update specs**
- **Update service defs**
- **added test for relation related upsert , fixed formatting**
- **Update specs**
- **Trigger CI**
- **Add user injection and permission handling in Upsert actions**
- **Fix merge**
- **Downgrade migrations**
- **Revert "Downgrade migrations"**
- **chore: regen specs**
- **chore: review comments**
- **chore: uncomment**
- **chore: use class**
- **framework support + content for OTP & add-member email**
- **chore: regen specs**
- **chore: client spec changes**
- **chore: update to return object**
- **handle case when there are multiple sdk methods**
- **Skip deployment when commit is created by us**
- **update commit author name param**
- **Add defaultBranch in getRepository response**
- **Fix index models**
- **Update lock**
- **fix: route didn't include the params needed.**
- **compress screenshots before uploading**
- **Switch from gzip to none compression**
- **add tests**
- **update composer**
- **chore: regen**
- **chore: regen**
- **chore: regen**
- **chore: remove**
- **chore: remove**
- **chore: regen**
- **regen sdks**
- **uncomment**
- **fix: compose**
- **chore: regen**
- **Preview texts for emails**
- **Suppress git-action exception in deployment worker**
- **fix: unwanted method being added to specs**
- **Better error message for invalid function scheduled time**
- **validate only when scheduledAt is available**
- **chore: update sdks 1.7.x**
- **update swift**
- **chore: add more changelog**
- **chore: add more changelog**
- **chore: update to use angle brackets**
- **send log to queue**
- **tests**
- **change error msg**
- **format**
- **remove dot**
- **docs: update CONTRIBUTING.md to clarify how to start**
- **feat: stats-usage on redis**
- **fix: templates on `1.7.x`.**
- **update version for rn**
- **Change preview & body for MFA email**
- **chore: add docs for nestedType, encode, from and toMap**
- **add docs for android as well**
- **add java example**
- **chore: add docs in web and node**
- **add docs in rn**
- **Update database**
- **chore: update go**
- **chore: use config.json**
- **fix: gitignore**
- **Fix merge**
- **Fix consumers**
- **update: move to grids api.**
- **fix: response modals and sdk-gen for multiple namespaces.**
- **fix: tests.**
- **fix: tests.**
- **update: graphql tests.**
- **update: docs.**
- **regen: specs and fix file name.**
- **delete: tables examples.**
- **fix: no params on multiple methods.**
- **update: regen specs.**
- **update: regen examples.**
- **update: regen examples for grids.**
- **remove: local setup.**
- **remove: local setup.**
- **update: deprecation methods.**
- **address comments.**
- **fix: tests.**
- **update: regen specs with updated deprecations.**
- **fix: deprecation inheritance on other methods with different namespace.**
- **regen: specs.**
- **Update migration**
- **reset: docs for easier review.**
- **reset: docs for easier review.**
- **reset: docs for easier review.**
- **Revert migrations, update db**
- **Downgrade db**
- **Update generator**
- **Run tests**
- **Use null default with fallback**
- **fallbacks**
- **Run tests**
- **use 0.73.1**
- **Filter sequence to int because any models skip rule checks**
- **Add test case**
- **updated migration release**
- **Update 0.73.2**
- **fix: create email target when using email otp registration**
- **linebreak**
- **Skip empty queries**
- **Update migrations**
- **formatting**
- **chore: filter certificates renewal task in maintenance by region**
- **chore: improve check**
- **chore: use md5 if enabled**
- **Remove sequence on incoming docs**
- **Add missing removal**
- **Fix CSV success test**
- **Add internal attributes CSV test**
- **remove: tables examples.**
- **update: regen examples, references.**
- **chore: remove console sdk from platforms**
- **chore: move changelog path to platforms config**
- **chore: add console back**
- **update: sdk-gen.**
- **Add permissions support for CSV**
- **Fix ref**
- **Fix tests**
- **Fix test**
- **feat: telemetry for cache hits and misses**
- **Revert composer**
- **chore: update to use gauge**
- **fix nesting if**
- **chore: update name**
- **chore: add total**
- **chore: update counter**
- **Update db**
- **Run tests**
- **Run tests**
- **added modify createdAt/updatedAt support**
- **Update db + migrations**
- **chore: update changelog and sdk gen**
- **fix: 500 errors on robots and humans txt files**
- **chore: fallback to main domain**
- **Restrict createdAt and updatedAt attributes for non-API key and non-privileged users in Create, Update, and Upsert actions**
- **feat: add model examples + additonal examples to specs**
- **chore: simplify**
- **Add comprehensive tests for date and time operations in database rows**
- **updated composer**
- **changed the error message for client side**
- **chore: update naming and specs**
- **chore: update metric naming**
- **chore: update sdk gen**
- **chore: add placeholder version**
- **Remove $sequence**
- **Pull main**
- **Run tests**
- **Run tests**
- **chore: add support for svg favicons**
- **chore: add svg sanitization**
- **add test**
- **fix: cleansvg**
- **chore: fix import**
- **chore: update test**
- **Check CAA record in DNSTest**
- **Verify domain CAA after certificate issuance**
- **Update test**
- **Add check in proxy**
- **Add default value**
- **Update console tests**
- **update 1.7.x examples**
- **Run tests**
- **Run tests**
- **chore: minify svg**
- **fix: minify**
- **fix: favicons endpoint to fallback to ico instead of throwing error**
- **Improve PR quality**
- **Update composer.lock**
- **Fix tests**
- **Improve docs**
- **feat: Enhance database operations with event queues and improve test coverage for bulk operations**
- **Support empty CAA records**
- **Fix caa validation breaking existing validations**
- **Simplify implementation**
- **feat: Initialize event and payload properties in Event class and optimize bulk document queue triggering**
- **feat: Enhance RealtimeCustomClientTest with permission updates and timeout assertions for clients**
- **refactor: Rename triggerQueuesForBulkDocuments to triggerBulk for consistency across document actions**
- **fix: Update getUser method call in RealtimeCustomClientTest for clarity**
- **Revert "Check CAA record before issuing certificate"**
- **Fix atomic num ops with limit 0**
- **Add 'type' attribute to database creation and update tests**
- **Fix caa records inheritance**
- **Update database 0.75.***
- **Revert revert of CAA validation**
- **updated migration release**
- **updated the validator of type of database to validator**
- **updaetd composer version**
- **updated database response model rules**
- **added migration to self hosted version**
- **removed $id to get updated**
- **replaced create attribute with a util createAttributeFromCollection method**
- **updated composer**
- **Fix refs**
- **Fix tests**
- **chore: update SDK generator and sdks**
- **Improve tests**
- **Formatting fix**
- **Fix manual QA edge case**
- **PR review fixes**
- **updated database type to grids and legacy**
- **upgraded composer for utopia migration**
- **chore: add readonly param to sequence, databaseId and collectionId**
- **fix: wrong resource path.**
- **fix: lint**
- **update: `grid` on resource events.**
- **update: readability.**
- **update: if/else > switch > match**
- **update casing**
- **Update migrations**
- **Add placeholder detection**
- **Fix formatting**
- **Fix tests**
- **Update base image to appwrite/base:0.10.3**
- **Fix file ownership syntax using colon instead of dot separator**
- **Update QR code avatar test to use fixed file size instead of file read**
- **Update credit card icon size in AvatarsTest**
- **Make functions worker customizable**
- **Fix typo**
- **Update build command for flutter**
- **Improve QR code image testing with pixel-level comparison**
- **Add assertSamePixels method to Scope class for image comparison**
- **Inline image comparison logic in Scope test class, remove ImageAssertions**
- **Update avatar test to validate response headers and content type**
- **Fix whitespace and indentation in AvatarsBase test method**
- **Update avatar test to validate response headers and content type**
- **Update GraphQL avatars test to validate response headers and content**
- **Refactor: Move assertSamePixels method to a different location in Scope class**
- **Fix tests**
- **chore: make check for adding warning header stricter**
- **Purge collection cache alongside document cache in database operations**
- **Add database query tests for notContains, notSearch, and other filters**
- **Skip redundant subqueries in users list route**
- **Bump flutter starter version**
- **Update DB**
- **Revert "Feat: Make functions worker customizable"**
- **Fix missing place not using proper publisher**
- **Reapply "Feat: Make functions worker customizable"**
- **Linter fix**
- **Fix flakyness**
- **Fix notSearch with no FTS**
- **Update tests**
- **Fix index count tests**
- **Fix not search on wrong attribute**
- **Update lock**
- **Fix tests**
- **Remove temp logging**
- **feat: allow HEAD requests and add test for it**
- **update client**
- **Implement CI/CD action to check EN locale**
- **Revert "Merge pull request #10299 from appwrite/feat-functions-redis-queue"**
- **Finalized locale check script**
- **Remove unused translations**
- **Add CI/CD job**
- **update executor**
- **Apply suggestion from @Copilot**
- **Apply suggestion from @Copilot**
- **Add fallback locale support + tests**
- **Fix CI/CD command**
- **Fix variables sharing across resources**
- **Update invite email copy**
- **chore: remove experiment warnings from bulk methods**
- **update specs**
- **Update versions**
- **chore: add method desc and demo**
- **Update audit lib & platform**
- **Update svg-sanitizer**
- **Update package**
- **revert lock**
- **update specs**
- **Add upsert event**
- **Fix uncaught invalid arg**
- **Fix test**
- **Fix events**
- **Port desc fixes**
- **Add internal file upload support and authorization checks in migrations**
- **Port inc/dec fixes**
- **Port tests**
- **Lint + lock**
- **Update 3rd aprty deps + fix tests**
- **Fix index delete event**
- **Add grids upsert event**
- **Revert "Fix index delete event"**
- **Add missing attribute/index update events**
- **Fix path**
- **updated the the description of the internal files**
- **Increase dynamic API key expiration**
- **fix - incorrect file token expiry**
- **linter**
- **Linter fix**
- **Improve ci/cd logging**
- **Fix tests**
- **Speed-up critical failures in assertEventuelly**
- **Add extensive build logging**
- **Improve flakyness**
- **formatting fix**
- **Improve font reliability**
- **Migrate to TablesDB**
- **Update workflow**
- **Namespace fix**
- **Lint**
- **Switch removable attributes**
- **Fix test paths**
- **Fix imports**
- **Test fixes**
- **Improve fonts**
- **Add usage routes**
- **Fix collection context**
- **Remove db multimethod**
- **Add log route**
- **Move usage**
- **TablesDB CRUD**
- **Update registry**
- **Update tests**
- **Fix context**
- **Fix spec gen**
- **Fix dangling tests**
- **Fix usage**
- **Fix index context**
- **Test fixes**
- **Fix tests**
- **Revert stop on failure**
- **Address stnguyen90 feedback: simplify JWT logic and fix test expectations**
- **Better jwt handling and tests**
- **Fix hardcoded collectionId**
- **database 0.71.15**
- **Remove filter**
- **Fix update/upsert set collection ID**
- **Remove invalid test**
- **Remove weird static/const mix**
- **Fix tablesDB GQL payloads for DB resource**
- **http fix to double slashes and new domain validator**
- **Revert "http fix to double slashes and new domain validator"**
- **Reapply "http fix to double slashes and new domain validator"**
- **Fix static user refs**
- **Fix un-fixed ref**
- **adjust tests to new domain validator**
- **Casing**
- **Fix references**
- **Fix specs**
- **Format**
- **Fix namespace**
- **Fix email gql**
- **Fix namespace**
- **Fix namespace**
- **More fix namespace**
- **Update specs**
- **Fix service refs**
- **Fix specs**
- **Pin framework to stable**
- **Fix backwards comptibility**
- **Fix attribute update refactor collision**
- **Update platforms**
- **Remove redundant param**
- **Add back dart example**
- **Fix lint**
- **Fix workflow**
- **Fix docs**
- **Specs**
- **Fix tests**
- **Fix inc/dec auth types**
- **Update specs**
- **Dart/flutter changelog/getting started cleanup**
- **Docs**
- **Remove bunnyCDN and Alpine external dependency**
- **Fix during QA**
- **PR review improvements**
- **Increase wait timeout**
- **Add executionId and client IP to execution headers**
- **Add solution for function domain**
- **Add async=false solution**
- **Truncate logs in function worker**
- **Fix docs refs**
- **Fix inline doc refs**
- **Fix base list doc desc**
- **Fix PR changes**
- **Fix acronym methods, deprecate old versions**
- **Update tablesDb -> tablesDB**
- **Update tests/e2e/Services/Tokens/TokensConsoleClientTest.php**
- **Lint**
- **update: user create events logic.**
- **Truncate logs and errors from beginning**
- **Test truncation in one test**
- **Update lock**
- **Add cookie test for site execution**
- **bump: lock.**
- **Update lock**
- **Enhance token JWT validation in TokensConsoleClientTest with comprehensive checks**
- **test fix/lint**
- **fix: publisher consumers for functions.**
- **revert: database listener changes.**
- **update: comment.**
- **Update app/controllers/general.php**
- **Add deploymentId to execution response model**
- **Formatting fix**
- **Fix tests**
- **Utilize assets server for fonts**
- **Update generator + specs**
- **Fix attribute refs**
- **Fix typo**
- **Fix deprecation messages**
- **Specs + doc regen**
- **Fix nothing was updated**
- **Run tests**
- **Check no change**
- **Restrict array list**
- **Restrict array list**
- **Update generator**
- **bump**
- **deploy: fix queue health**
- **Update lock**
- **chore: allow publisherMessaging override**
- **Sync docs**
- **Update base image**
- **Address PR comments**
- **chore: switch Union China Pay to just Union Pay**
- **update specs**
- **Replace x-real-ip with ->getIP()**
- **Update docs**
- **Apply improved domain validator**
- **fix tests**
- **Add new tests**
- **PR review changes**
- **chore: add replacewith and deprecated since to updatePhoneSession and updateMagicURLSession**
- **fix: version, remove boolean**
- **format**
- **remove instance of**
- **Add execution id and log id to response headers**
- **Update specs**
- **Fix demo paths**
- **Remove execution headers from allowlist**
- **Update function domain test**
- **Update CLI version**
- **Fix spaces in templates**
- **Fix request filter for selections**
- **chore: update union china pay to just unionpay**
- **chore: update specs**
- **fomat**
- **Initial plan**
- **Apply sequence integer casting fix to Row.php and add test assertion**
- **Add $tenant attribute removal to Action.php processDocument method**
- **fix: Remove $sequence attribute from documents on create and update**
- **chore: fix lint error**
- **feat: allow HEAD requests in function executions**
- **format**
- **Fix domain validator**
- **Array scopes**
- **Remove response filter**
- **Format**
- **Fix GQL test**
- **Fix relationship request filter**
- **Bump console**
- **Format**
- **Ignore relationships that are not available**
- **Update CLI**
- **Fix schedule execution test**
- **Update default method in description**
- **update specs**
- **add test case**
- **Fix request filter for nested relationships**
- **fix: 201**
- **delete execution**
- **Remove logging**
- **Update desc**
- **Update specs**
- **Format**
- **Update src/Appwrite/Utopia/Request/Filters/V20.php**
- **chore: update tables group in specs**
- **fix: namespace naming from tables-db to tablesdb in specs**
- **use single quotes**
- **chore: update description for upsert methods**
- **casing**
- **added processing of migration resource stats**
- **lint**
- **updated migration version**
- **updated reset of the stats usage worker to reset metrics**
- **linting**
- **optimised the memory usage by resource aggregation**
- **Update src/Appwrite/Platform/Workers/Migrations.php**
- **refactor: prevent direct publisher calls**

### CODE REVIEW

File: app/controllers/api/health.php [hunk 1]
```diff
@@ -950,11 +960,53 @@
...
+        Database $queueForAudits,
+        Database $queueForMails,
+        Database $queueForFunctions,
+        Database $queueForStatsResources,
+        Database $queueForStatsUsage,
+        Database $queueForWebhooks,
+        Database $queueForCertificates,
+        Database $queueForBuilds,
+        Database $queueForMessaging,
+        Database $queueForMigrations
```
Comment (abnegate) [inline app/controllers/api/health.php]: Wrong types

File: src/Appwrite/Event/Event.php [hunk 2]
```diff
@@ -638,4 +638,39 @@ private static function mirrorCollectionEvents(string $pattern, string $firstEve
...
         return $events;
     }
+
+    /**
+     * Clone the event instance.
+     *
+     * Creates a deep copy of the event with all properties,
+     * including nested objects and arrays.
+     */
+    public function __clone(): void
```
Comment (abnegate) [inline src/Appwrite/Event/Event.php]: With this added do we still need the `from` method?

File: src/Appwrite/Event/Func.php [hunk 3]
```diff
@@ -219,4 +219,23 @@ protected function preparePayload(): array
...
+        parent::__clone();
+
+        if ($this->function !== null) {
+            $this->function = clone $this->function;
+        }
+
+        if ($this->execution !== null) {
+            $this->execution = clone $this->execution;
+        }
+    }
```
Comment (abnegate) [inline src/Appwrite/Event/Func.php]: Some other queues have objects that will need cloned, databases at the least

File: src/Appwrite/Platform/Tasks/ScheduleExecutions.php [hunk 4]
```diff
@@ -26,12 +29,29 @@ public static function getCollectionId(): string
         return 'executions';
     }
 
+    public function __construct()
+    {
+        $type = static::getSupportedResource();
```
Comment (abnegate) [inline src/Appwrite/Platform/Tasks/ScheduleExecutions.php]: Let's set this in the parent constructor and call parent::__construct here

File: src/Appwrite/Platform/Tasks/ScheduleExecutions.php [hunk 5]
```diff
@@ -26,12 +29,29 @@ public static function getCollectionId(): string
...
+
+        $this
+            ->desc("Execute executions scheduled in Appwrite")
```
Comment (abnegate) [inline src/Appwrite/Platform/Tasks/ScheduleExecutions.php]: Single quotes

File: src/Appwrite/Platform/Tasks/ScheduleFunctions.php [hunk 6]
```diff
@@ -31,6 +32,23 @@ public static function getCollectionId(): string
         return 'functions';
     }
 
+    public function __construct()
+    {
+        $this
+            ->desc("Execute functions scheduled in Appwrite")
```
Comment (abnegate) [inline src/Appwrite/Platform/Tasks/ScheduleFunctions.php]: Single quotes

File: src/Appwrite/Platform/Tasks/ScheduleMessages.php [hunk 7]
```diff
@@ -25,6 +28,23 @@ public static function getCollectionId(): string
         return 'messages';
     }
 
+    public function __construct()
+    {
+        $this
+            ->desc("Execute messages scheduled in Appwrite")
```
Comment (abnegate) [inline src/Appwrite/Platform/Tasks/ScheduleMessages.php]: Single quotes

File: src/Appwrite/Event/Event.php (hunk 2 above)
Comment (loks0n) [reply src/Appwrite/Event/Event.php]: I think so, for example here https://github.com/appwrite/appwrite/blob/bcd5cab98fa082ec288d8504bf5384ed85f6130d/src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Documents/Action.php#L352

seems like a different use.

File: src/Appwrite/Platform/Tasks/ScheduleFunctions.php (hunk 6 above)
Comment (loks0n) [reply src/Appwrite/Platform/Tasks/ScheduleFunctions.php]: fixed

File: src/Appwrite/Platform/Tasks/ScheduleMessages.php (hunk 7 above)
Comment (loks0n) [reply src/Appwrite/Platform/Tasks/ScheduleMessages.php]: fixed

File: src/Appwrite/Platform/Tasks/ScheduleExecutions.php (hunk 5 above)
Comment (loks0n) [reply src/Appwrite/Platform/Tasks/ScheduleExecutions.php]: fixed

File: src/Appwrite/Platform/Tasks/ScheduleExecutions.php (hunk 4 above)
Comment (loks0n) [reply src/Appwrite/Platform/Tasks/ScheduleExecutions.php]: This was unused, removed it

File: app/controllers/api/health.php (hunk 1 above)
Comment (loks0n) [reply app/controllers/api/health.php]: fixed

File: src/Appwrite/Event/Func.php (hunk 3 above)
Comment (loks0n) [reply src/Appwrite/Event/Func.php]: Added database event clone, do you think I should implement for all of them?

File: app/controllers/api/health.php [hunk 8]
```diff
@@ -950,11 +960,53 @@
...
+        Build $queueForBuilds,
+        Messaging $queueForMessaging,
+        Migration $queueForMigrations
+    ) {
         $threshold = \intval($threshold);
 
-        $failed = $publisher->getQueueSize(new Queue($name), failedJobs: true);
+        /** @var Event $queue */
+        $queue = match ($name) {
+            Event::DATABASE_QUEUE_NAME => $queueForDatabase,
```
Comment (ChiragAgg5k) [inline app/controllers/api/health.php:996]: we are kind of repeating this here in health checkpoint whitelist as well, do you think it would be better to have a method called getQueueNames() in Event class so we dont have repeat this in two places?

Image

**Review by ChiragAgg5k — APPROVED** [review]

File: app/controllers/api/health.php (hunk 8 above)
Comment (loks0n) [reply app/controllers/api/health.php:996]: I actually already don't like that we put the constants here, I think this should be a polymorphic field of each different event, so I'd rather add more ways to consume this
